diff --git a/MODULE.bazel b/MODULE.bazel
index 16776ba..683c131 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -70,7 +70,7 @@ deb = use_repo_rule("@download_utils//download/deb:defs.bzl", "download_deb")
 deb(
     name = "acl-deb",
     build = "//third_party/acl:acl.BUILD",
-    urls = ["https://archive.ubuntu.com/ubuntu/pool/main/a/acl/libacl1-dev_2.2.52-3build1_amd64.deb"],
+    urls = ["https://launchpadlibrarian.net/581258948/libacl1-dev_2.3.1-1_arm64.deb"],
     visibility = ["//visibility:public"],
 )
 
diff --git a/third_party/acl/acl.BUILD b/third_party/acl/acl.BUILD
index 718726c..710befd 100644
--- a/third_party/acl/acl.BUILD
+++ b/third_party/acl/acl.BUILD
@@ -1,7 +1,7 @@
 cc_library(
     name = "acl",
     srcs = [
-        "usr/lib/libacl.a",
+        "usr/lib/aarch64-linux-gnu/libacl.a",
     ],
     hdrs = [
         "usr/include/acl/libacl.h",
@@ -9,4 +9,4 @@ cc_library(
     ],
     includes = ["usr/include/"],
     visibility = ["//visibility:public"],
-)
+)
